### ISSUE TITLE
VB compiler incorrectly treats a parenthesized receiver as a variable

### ISSUE THREAD

**Issue by gafter** [dotnet/vblang#51]
See https://github.com/dotnet/roslyn/issues/4687 for details

**Comment by AnthonyDGreen** [comment]
@gafter my reading of dotnet/roslyn#4687 is that the break from the native compiler was fixed and that the Roslyn compiler now correctly treats a parenthesized expression as an r-value. If the request is that the specification say this then https://github.com/dotnet/vblang/blob/master/spec/expressions.md#parenthesized-expressions explicitly states this.

Is my understanding of the request correct and may I close?

**Comment by gafter** [comment]
@AnthonyDGreen No, that is not correct. The following program prints `1 2`, when according to the specification it must print `1 1`. Either the compiler must change or the specification must change.

``` vb
Module Module1

    Sub Main()
        Dim s = New S
        Call s.SetField(1)
        Console.WriteLine(s.Field) ' should print 1
        Call (s).SetField(2)
        Console.WriteLine(s.Field) ' should print 1
    End Sub

End Module

Structure S
    Public Field As Integer
    Public Sub SetField(Value As Integer)
        Me.Field = Value
    End Sub
End Structure
```
